# Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.


In [2]:
import numpy as np

# Create two arrays of data for studying time and exam scores
studying_time = np.array([10, 8, 9, 7, 6, 10, 5, 7, 8, 6])
exam_scores = np.array([90, 85, 89, 80, 78, 93, 70, 82, 87, 75])

# Calculate the Pearson correlation coefficient
r = np.corrcoef(studying_time, exam_scores)[0,1]

# Print the result
print("Pearson correlation coefficient:", r)


Pearson correlation coefficient: 0.971948390231306


the Pearson correlation coefficient is close to +1, it indicates a strong positive linear relationship between studying time and exam scores. In this case, we could predict that as students spend more time studying, their exam scores are likely to increase.

# Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.


In [5]:
from scipy.stats import spearmanr

# Create two lists of data for sleep and job satisfaction
sleep = [7, 6, 8, 5, 7, 6, 8, 7, 6, 5]
job_satisfaction = [8, 5, 9, 6, 7, 5, 8, 7, 6, 5]

# Calculate the Spearman's rank correlation
rho,pvalue = spearmanr(sleep, job_satisfaction)

# Print the result
print("Spearman's rank correlation coefficient:", rho)



Spearman's rank correlation coefficient: 0.8626594336638181


The Spearman's rank correlation coefficient is close to +1, it indicates a strong monotonic relationship between the two variables, meaning that as the amount of sleep individuals get each night increases, so does their job satisfaction level.

# Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.


In [12]:
import pandas as pd
from scipy.stats import pearsonr, spearmanr

# Create a dataframe with the data
data = pd.DataFrame({'hours_exercise': [4, 5, 2, 6, 3, 4, 5, 3, 2, 4, 
                                        5, 3, 4, 6, 2, 3, 5, 4, 2, 3, 
                                        4, 5, 3, 6, 2, 4, 5, 3, 2, 4, 
                                        5, 3, 4, 6, 2, 3, 5, 4, 2, 3, 
                                        4, 5, 3, 6, 2, 4, 5, 3, 2, 4],
                     'bmi': [23, 27, 32, 29, 28, 25, 24, 30, 33, 27,
                             26, 28, 31, 28, 26, 29, 24, 27, 31, 30,
                             28, 25, 30, 27, 29, 26, 24, 31, 28, 30,
                             26, 29, 27, 25, 30, 31, 28, 24, 26, 29,
                             30, 27, 28, 24, 31, 26, 29, 27, 30, 28]})

# Calculate the Pearson correlation coefficient
pearson_corr, pval_pearson = pearsonr(data['hours_exercise'], data['bmi'])

# Calculate the Spearman's rank correlation coefficient
spearman_corr, pval_spearman = spearmanr(data['hours_exercise'], data['bmi'])

# Print the results

print("Pearson correlation coefficient:", pearson_corr)
print("Spearman's rank correlation coefficient:", spearman_corr)



Pearson correlation coefficient: -0.5506437049279406
Spearman's rank correlation coefficient: -0.5620554532189472


The Pearson correlation coefficient measures the linear relationship between two variables, while the Spearman's rank correlation coefficient measures the monotonic relationship. If the relationship between the two variables is approximately linear, the Pearson correlation coefficient is likely to be more appropriate. If the relationship is non-linear, the Spearman's rank correlation coefficient may be more appropriate. It's also possible that the two coefficients will give similar results, in which case you can choose the one that makes more sense in the context of your study

# Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.


In [13]:
import pandas as pd
from scipy.stats import pearsonr

# Create a dataframe with the data
data = pd.DataFrame({'hours_tv': [3, 4, 2, 6, 3, 4, 5, 3, 2, 4,
                                   5, 3, 4, 6, 2, 3, 5, 4, 2, 3,
                                   4, 5, 3, 6, 2, 4, 5, 3, 2, 4,
                                   5, 3, 4, 6, 2, 3, 5, 4, 2, 3,
                                   4, 5, 3, 6, 2, 4, 5, 3, 2, 4],
                     'activity_level': [2, 4, 3, 6, 4, 3, 5, 4, 3, 4,
                                        6, 5, 4, 5, 3, 4, 6, 4, 3, 4,
                                        5, 4, 6, 4, 3, 5, 6, 4, 3, 4,
                                        5, 4, 6, 4, 3, 5, 6, 4, 3, 4,
                                        5, 4, 6, 4, 3, 5, 6, 4, 3, 4]})

# Calculate the Pearson correlation coefficient
pearson_corr, pval = pearsonr(data['hours_tv'], data['activity_level'])

# Print the results
print("Pearson correlation coefficient:", pearson_corr)


Pearson correlation coefficient: 0.5730525162741855


A Pearson correlation coefficient of 0.573 indicates a moderate positive correlation between the number of hours individuals spend watching television per day and their level of physical activity. This means that as the number of hours spent watching TV increases, so does the level of physical activity to some extent.

# Q5. A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:

In [16]:
import pandas as pd
df = pd.DataFrame({"Age(years)":[25,42,37,19,31,28],"Soft Drink Preference":["Coke","Pepsi","Mountain Dew","Coke","Pepsi","Coke"]})
df

Age(years) Soft Drink Preference
0          25                  Coke
1          42                 Pepsi
2          37          Mountain Dew
3          19                  Coke
4          31                 Pepsi
5          28                  Coke

In [20]:
import pandas as pd
from scipy.stats import chi2_contingency

# Create a dataframe with the data
df = pd.DataFrame({"Age(years)":[25,42,37,19,31,28],"Soft Drink Preference":["Coke","Pepsi","Mountain Dew","Coke","Pepsi","Coke"]})

# Create a contingency table
contingency_table = pd.crosstab(df['Age(years)'], df['Soft Drink Preference'])

# Print the contingency table
print(contingency_table)

# Perform the chi-square test of independence
chi2_stat, pval, dof, expected = chi2_contingency(contingency_table)

# Print the results
print("Chi-square test statistic:", chi2_stat)
print("p-value:", pval)
print("Degrees of freedom:", dof)
print("Expected frequencies:\n", expected)



Soft Drink Preference  Coke  Mountain Dew  Pepsi
Age(years)                                      
19                        1             0      0
25                        1             0      0
28                        1             0      0
31                        0             0      1
37                        0             1      0
42                        0             0      1
Chi-square test statistic: 12.000000000000002
p-value: 0.28505650031663116
Degrees of freedom: 10
Expected frequencies:
 [[0.5        0.16666667 0.33333333]
 [0.5        0.16666667 0.33333333]
 [0.5        0.16666667 0.33333333]
 [0.5        0.16666667 0.33333333]
 [0.5        0.16666667 0.33333333]
 [0.5        0.16666667 0.33333333]]


# Q6. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [21]:
import pandas as pd
from scipy.stats import pearsonr

# Create a dataframe with the data
data = pd.DataFrame({'sales_calls': [30, 50, 20, 60, 40, 30, 70, 80, 40, 50,
                                     60, 30, 40, 50, 20, 30, 70, 50, 20, 30,
                                     40, 60, 30, 70, 20, 50, 70, 30, 20, 40],
                     'sales': [5, 8, 3, 10, 6, 4, 12, 13, 7, 8,
                               10, 5, 6, 8, 3, 5, 11, 8, 2, 4,
                               6, 9, 4, 12, 2, 8, 12, 5, 3, 7]})

# Calculate the Pearson correlation coefficient
pearson_corr, pval = pearsonr(data['sales_calls'], data['sales'])

# Print the results
print("Pearson correlation coefficient:", pearson_corr)

Pearson correlation coefficient: 0.9897938260254878


A positive Pearson correlation coefficient indicate a positive correlation between the number of sales calls made per day and the number of sales made per week